### PR TITLE
Use IdCompressor for RevisionTag

### PR DESCRIPTION
**Pull request by justus-camp**
## Description
Instead of generating UUIDs for the RevisionTag, we're now going to use the runtime's IdCompressor to generate Ids. This implementation translates the RevisionTag to an OpSpaceCompressedId before sending an op and back to a SessionSpaceCompressedId when receiving an op. 

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -1319,6 +1315,10 @@ export class ContainerRuntime
 
 		const pendingRuntimeState = pendingLocalState as IPendingRuntimeState | undefined;
 
+		if (this.idCompressorEnabled) {
```
Comment (kian-thompson) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1318]: Maybe add a comment if call order is important here

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -2216,11 +2202,19 @@ export class ContainerRuntime
 				this.blobManager.processBlobAttachOp(messageWithContext.message, local);
 				break;
 			case ContainerMessageType.IdAllocation:
+				console.log(messageWithContext.message.contents);
```
Comment (kian-thompson) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2205]: console.log

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -3377,13 +3371,13 @@ export class ContainerRuntime
...
 					metadata: undefined,
-					localOpMetadata: this.idCompressor?.serialize(true),
+					localOpMetadata: undefined,
 					type: ContainerMessageType.IdAllocation,
 				};
 			}
 
 			if (idAllocationBatchMessage !== undefined) {
-				this.outbox.submit(idAllocationBatchMessage);
+				this.outbox.submitIdAllocation(idAllocationBatchMessage);
```
Comment (kian-thompson) [inline packages/runtime/container-runtime/src/containerRuntime.ts:3380]: nit: I don't think you need this `idAllocationBatchMessage` and could move the `submitIdAllocation(...)` call into the `if` scope above

**Review by kian-thompson — COMMENTED** [review]
Looks good to me overall, but maybe @wes-carlson can have another look since this deals with stashed ops.

File: experimental/dds/tree2/src/test/feature-libraries/sequence-field/compose.spec.ts [hunk 4]
```diff
@@ -1013,9 +1014,15 @@ describe("SequenceField - Compose", () => {
 			[0, 1, 2],
 			[2, 1, 0],
 		]) {
-			const move1 = tagChange(Change.move(a, 1, b > a ? b + 1 : b), tag1);
-			const move2 = tagChange(Change.move(b, 1, c > b ? c + 1 : c), tag2);
-			const return2 = tagRollbackInverse(
+			const move1: TaggedChange<SF.Changeset<never>> = tagChange(
```
Comment (justus-camp) [inline experimental/dds/tree2/src/test/feature-libraries/sequence-field/compose.spec.ts:1017]: Not sure why I had to add these type annotations. After my changes I was suddenly getting an error here and had to explicitly specify the type. You'll see this change in a good number of files.

**Pull request closed without merging**